server/src/main/java/org/opensearch/search/backpressure/settings/SearchBackpressureSettings.java

```diff
@@ -63,6 +63,11 @@ private static class Defaults {
         Defaults.CANCELLATION_RATIO,
         0.0,
         1.0,
+        value -> {
```

Comment:
This validation looks redundant, the `Setting.doubleSetting()` has min and max bounds that are set to `0.0`and `1.0`  respectively, didn't validation trigger?

Reply:
Yes, the problem is that 0.0 is not acceptable, users can set any value greater than 0.0, but no minimal bound exists.

Reply:
In this case we have three choices:
 - make sure 0.0 is handled properly down the line
 - have reasonable minimal value (fe. 0.01)
 - use different factory method that does use custom validation without min/max bounds

Reply:
Choice 3 makes more sense, please take a look at the new change, thanks!